src/xrc/src/forex.rs

```diff
@@ -945,11 +957,51 @@ mod test {
         assert!(maybe_rate.is_none());
     }
 
+    // A helper function that adds enough rates to [ForexRateStore] so that get calls succeed in tests
+    fn add_enough_cxdr_rates_to_store(rates_store: &mut ForexRateStore, timestamp: u64) {
+        let mut rates = Vec::<u64>::new();
+        for _i in 0..MIN_SOURCES_TO_REPORT {
+            rates.push(800_000_000);
+        }
+        rates_store.put(
+            timestamp,
+            hashmap! {
+                COMPUTED_XDR_SYMBOL.to_string() =>
+                    QueriedExchangeRate::new(
+                        Asset {
+                            symbol: COMPUTED_XDR_SYMBOL.to_string(),
+                            class: AssetClass::FiatCurrency,
+                        },
+                        Asset {
+                            symbol: USD.to_string(),
+                            class: AssetClass::FiatCurrency,
+                        },
+                        timestamp,
+                        &rates,
+                        MIN_SOURCES_TO_REPORT,
+                        MIN_SOURCES_TO_REPORT,
+                        Some(timestamp),
+                    )
+            },
+        );
+    }
+
+    #[test]
+    fn rate_store_get_works_with_sufficient_cxdr_rates() {
+        let mut store = ForexRateStore::new();
+        add_enough_cxdr_rates_to_store(&mut store, 1234);
+        let result = store.get(1234, 1234, COMPUTED_XDR_SYMBOL, USD);
+        assert!(
+            matches!(result, Ok(rate) if rate.base_asset_num_received_rates == MIN_SOURCES_TO_REPORT)
+        );
+    }
+
     /// Tests that the [ForexRatesStore] struct correctly updates rates for the same timestamp.
     #[test]
     fn rate_store_update() {
```

Comment:
Can you also use the `QueriedExchangeRate` constructor in this test?

Reply:
Done for all usages in `forex.rs` and `api/test.rs`